tcms/telemetry/static/telemetry/js/testing/execution-dashboard.js

```diff
@@ -16,23 +16,43 @@ function preProcessData (data, callbackF) {
 
     jsonRPC('TestCase.filter', { pk__in: caseIds }, function (cases) {
         const testerPerCase = {}
+        const componentsPerCase = {}
         cases.forEach(function (element) {
             testerPerCase[element.id] = element.default_tester__username
         })
 
-        jsonRPC('TestRun.filter', { pk__in: runIds }, function (runs) {
-            const testerPerRun = {}
-            runs.forEach(function (element) {
-                testerPerRun[element.id] = element.default_tester__username
+        jsonRPC('Component.filter', { cases__in: cases.map(i => i.id) }, function (components) {
```

Comment:
`cases.map()` is the same as `casesIds` above. No need to calculate it again.

Reply:
note to self: caseIds and runIds above don't appear to filter out duplicate values. Need to clean this up.